component/src/main/java/org/wso2/extension/siddhi/map/avro/sourcemapper/AvroSourceMapper.java

```
@@ -0,0 +1,424 @@
+        this.streamAttributes = this.streamDefinition.getAttributeList();
+        this.streamAttributesSize = this.streamDefinition.getAttributeList().size();
+        schema = getAvroSchema(optionHolder.validateAndGetStaticValue(DEFAULT_AVRO_MAPPING_PREFIX.concat(".").
+                        concat(SCHEMA_IDENTIFIER),
+                null));
+    }
+
+    private Schema getAvroSchema(String schemaDefinition) {
+        if (schemaDefinition != null) {
+            return new Schema.Parser().parse(schemaDefinition);
+        } else {
+            log.error("Avro Schema is not specified in the stream definition");
+            return null;
+        }
+    }
+
+    /**
+     * Receives an event or events as a byte[] from source, converts it to
+     * a {@link org.wso2.siddhi.core.event.ComplexEventChunk}.
+     *
+     * @param eventObject       the input, given as a byte array
+     * @param inputEventHandler input handler
+     */
+    @Override
+    protected void mapAndProcess(Object eventObject, InputEventHandler inputEventHandler) throws InterruptedException {
+        Event[] convertedEvent = null;
+        try {
+            convertedEvent = convertToEvents(eventObject);
+        } catch (Throwable t) {
+            log.error("Exception occurred when converting Avro message: " + eventObject.toString() +
+                    " to Siddhi Event", t);
+        }
+
+        if (convertedEvent != null && convertedEvent.length > 0) {
+            inputEventHandler.sendEvents(convertedEvent);
+        }
+    }
+
+    private Event[] convertToEvents(Object eventObject) {
+        Object jsonObj = null;
+        Object avroObj;
+        String avroMessage;
+
+        if (eventObject instanceof byte[]) {
+            try {
+                jsonObj = AvroMessageProcessor.deserializeByteArray((byte[]) eventObject, schema);
+            } catch (Throwable t) {
+                log.error("Error when converting avro message of schema: " + schema.toString() +
```

IMO we can throw the error to the top level and handle?

---

We need to throw an error and drop the event and continue with the next event. So it is not necessary to throw it to top level.